benchmarks/performancepoetry/src/main/java/com/squareup/benchmarks/performance/poetry/PerformancePoemWorkflow.kt

```diff
@@ -0,0 +1,244 @@
+        }
+
+        if (currentStateIsLoading) {
+          context.runningWorker(
+            Worker.from {
+              delay(simulatedPerfConfig.complexityDelay)
+              stanzaIndex
+            }
+          ) {
+            action {
+              (state as? ComplexCall)?.let { currentState ->
+                state = Selected(currentState.payload)
+              }
+            }
+          }
+        }
+
+        val previousStanzas: List<StanzaRendering> =
+          if (stanzaIndex == NO_SELECTED_STANZA) emptyList()
+          else renderProps.stanzas.subList(0, stanzaIndex)
+            .mapIndexed { index, _ ->
+              context.renderChild(StanzaWorkflow, Props(renderProps, index), "$index") {
+                noAction()
+              }
+            }
+
+        val visibleStanza =
+          if (stanzaIndex == NO_SELECTED_STANZA) {
+            null
+          } else {
+            context.renderChild(
+              StanzaWorkflow, Props(renderProps, stanzaIndex), "$stanzaIndex"
+            ) {
+              when (it) {
+                CloseStanzas -> ClearSelection(simulatedPerfConfig)
+                ShowPreviousStanza -> SelectPrevious(simulatedPerfConfig)
+                ShowNextStanza -> SelectNext(simulatedPerfConfig)
+              }
+            }
+          }
+
+        val stackedStanzas = visibleStanza?.let {
+          (previousStanzas + visibleStanza).toBackStackScreen<Any>()
+        }
+
+        val stanzaListOverview =
+          context.renderChild(
+            StanzaListWorkflow,
+            renderProps
+          ) { selected ->
+            HandleStanzaListOutput(simulatedPerfConfig, selected)
+          }
+            .copy(selection = stanzaIndex)
+
+        isLoading.value = currentStateIsLoading
+        stackedStanzas
+          ?.let {
+            OverviewDetailScreen(
+              overviewRendering = BackStackScreen(stanzaListOverview),
+              detailRendering = it
+            )
+          } ?: OverviewDetailScreen(
+          overviewRendering = BackStackScreen(stanzaListOverview),
+          selectDefault = {
+            context.actionSink.send(HandleStanzaListOutput(simulatedPerfConfig, 0))
+          }
+        )
+      }
+    }
+  }
+
+  override fun snapshotState(state: State): Snapshot? = null
+
+  internal sealed class Action : WorkflowAction<Poem, State, ClosePoem>() {
+    abstract val simulatedPerfConfig: SimulatedPerfConfig
+
+    class ClearSelection(override val simulatedPerfConfig: SimulatedPerfConfig) : Action()
+    class SelectPrevious(override val simulatedPerfConfig: SimulatedPerfConfig) : Action()
+    class SelectNext(override val simulatedPerfConfig: SimulatedPerfConfig) : Action()
+    class HandleStanzaListOutput(
+      override val simulatedPerfConfig: SimulatedPerfConfig,
+      val selection: Int
+    ) : Action()
+
+    class ExitPoem(override val simulatedPerfConfig: SimulatedPerfConfig) : Action()
+
+    override fun Updater.apply() {
```

Comment:
Are you sure you can't move the `isLoading.value =` calls here? The first one could move to `initialState`.